### DISCUSSION TITLE
[V3 Pagination] How to remove the page from the query url ?

### DISCUSSION THREAD

**Discussion by timmy78** [livewire/livewire#7682] [1. Help]
Hello

I'm working with a component that has a pagination (WithPagination).

The query param "?page=2" is currently appearing, but in some cases I would like to remove it from the url.

**Was is the better way to do it ?** I find nothing about this in the documentation.

In the V2 I was overwritting `queryStringWithPagination` : 
```
public function queryStringWithPagination(): array
{
    $queryString = $this->traitQueryStringWithPagination();
    if (...) {
        unset($queryString['page']);
    }

    return $queryString;
}
```

After somes searches, I found that `\Livewire\Features\SupportPagination\SupportPagination::addUrlHook()` automatically add an attribute `PaginationUrl` to the component.

I'm actually doing that in my component : 

```
$this->attributes = $this->attributes->reject(function(Attribute $attribute) {
  return $attribute->getName() == 'paginators.page';
});
```

Thanks !

**Comment by joshhanley** [comment]
@timmy78 I'm not sure if there is an easy way at this stage to do it. Are you able to provide some more context around why/when you wouldn't want to show the pagination details in querystring? Thanks!

**Reply by timmy78** [reply]
My livewire component manages 2 comportments for the pagination : 
1. Normal pagination with the page in the url query
2. Infinite scroll with pagination. No html pagination, and the user cannot go directly to next pages.

An example would be to add a boolean function in the trait `HandlesPagination`, and check this one in `\Livewire\Features\SupportPagination\SupportPagination` before calling `addUrlHook()`

Thanks for the answer 😃

**Comment by BataBoom** [comment]
Hello,

I have a similar issue. Though I cant use your paginate method for multiple paginators cause I want to wrap/collect all results inside its own collection. Instead of assigning each paginator its own render variable. I've got it to work by making a new collection macro, but I figure it would break things like this. Though, I think I can fix this too but I cant figure out how to assign multiple Attributes to a variable. 

#[Validate('required|min:3')]
#[Url]
public string $search = '';

When I reset/refresh the page now the paginate query strings could remain as I can only use search as URL not validation. Then a new search string appends itself after the old pagination data. I made a gist showing the collection macro and full Livewire V3 Paginate class for assistance. 

https://gist.github.com/BataBoom/06942536e83c3ab05af3176760b7c903

Sorry if I hijacked this thread a bit, I believe its relatable and recent.
